Bottom: 52f54a9066c43779ebf2ff7fe6e106660e78d754
Top:    2a7eacc1e6c09814fbee6b15d25e417561dd2ff4
Author: Alexey Lyashkov <shadow@Alexeys-MacBook-Pro.local>
Date:   2016-01-20 08:15:08 +0300

Refresh of srv.patch

---

diff --git a/ib-sock.c b/ib-sock.c
index 5abf9a5..1816baa 100644
--- a/ib-sock.c
+++ b/ib-sock.c
@@ -240,8 +240,17 @@ struct IB_SOCK *ib_socket_create()
 
 void ib_socket_destroy(struct IB_SOCK *sock)
 {
+	struct IB_SOCK *pos, *next;
+
 	printk("IB socket destroy %p\n", sock);
 
+	sock->is_flags |= SOCK_ERROR;
+
+	list_for_each_entry_safe(pos, next, &sock->is_child, is_child) {
+		list_del(&pos->is_child);
+		ib_socket_destroy(pos);
+	}
+
 	if (sock->is_id)
 		rdma_destroy_id(sock->is_id);
 	kfree(sock);
